subsys/bootloader/bl_validation/bl_validation.c

```diff
@@ -267,20 +331,32 @@ static bool validate_firmware(u32_t fw_dst_address, u32_t fw_src_address,
 		return false;
 	}
 
-	if (!(((u32_t)fwinfo >= fw_src_address)
-		&& (((u32_t)fwinfo + fwinfo->total_size)
-			< (fw_src_address + fwinfo->size)))) {
+	if ((fwinfo->size > (PM_S0_SIZE))
+		|| (fwinfo->size > (PM_S1_SIZE))
```

Comment:
Why use S1 here and S0 above? The roles might be switched, so IMO we should assume that the size is symmetric, and maybe have a static assert here to highlight this.

Reply:
Added a static assert. and removed one check